src/service/passport/types.ts

```diff
@@ -0,0 +1,70 @@
+import { JwtPayload } from 'jsonwebtoken';
+
+export type JwkKey = {
+  kty: string;
+  kid: string;
+  use: string;
+  n?: string;
+  e?: string;
+  x5c?: string[];
+  [key: string]: any;
+};
+
+export type JwksResponse = {
+  keys: JwkKey[];
+};
+
+export type JwtValidationResult = {
+  verifiedPayload: JwtPayload | null;
+  error: string | null;
+};
+
+/**
+ * The JWT role mapping configuration.
+ *
+ * The key is the in-app role name (e.g. "admin").
+ * The value is a pair of claim name and expected value.
+ *
+ * For example, the following role mapping will assign the "admin" role to users whose "name" claim is "John Doe":
+ *
+ * {
+ *   "admin": {
+ *     "name": "John Doe"
+ *   }
+ * }
+ */
+export type RoleMapping = Record<string, Record<string, string>>;
+
+export type AD = {
+  isUserMemberOf: (
+    username: string,
+    groupName: string,
+    callback: (err: Error | null, isMember: boolean) => void,
+  ) => void;
+};
+
+/**
+ * The UserInfoResponse type from openid-client (to fix some type errors)
```

Comment:
Some details as to what the errors were would help with review

Reply:
I found a fix for this one by simply importing the `UserInfoResponse` from `openid-client`. I thought it came from `openid-client/passport` so I didn't try importing it directly.